GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -22,26 +23,63 @@ public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool forceRu
 
         public override string Area => nameof(LooseObjectsStep);
 
+        public virtual int MaxLooseObjectsInPack => 1000;
+
         protected override string LastRunTimeFilePath => Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjectsLastRunFileName);
         protected override TimeSpan TimeBetweenRuns => TimeSpan.FromDays(7);
 
         public int CountLooseObjects()
         {
             int count = 0;
-            
+
             foreach (string directoryPath in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
             {
                 string directoryName = directoryPath.TrimEnd(Path.DirectorySeparatorChar).Split(Path.DirectorySeparatorChar).Last();
 
                 if (GitObjects.IsLooseObjectsDirectory(directoryName))
-                {                    
+                {
                     count += this.Context.FileSystem.GetFiles(Path.Combine(this.Context.Enlistment.GitObjectsRoot, directoryPath), "*").Count();
                 }
             }
 
             return count;
         }
 
+        public void PackLooseObjects(StreamWriter streamWriter)
+        {
+            // Find looseObjects
+            foreach (string directoryPath in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
+            {
+                string directoryName = directoryPath.TrimEnd(Path.DirectorySeparatorChar).Split(Path.DirectorySeparatorChar).Last();

```

Comment:
`System.IO.Directory.GetParent()`?

Reply:
or rather, `Path.GetDirectory()`?

Reply:
Are you suggesting instead of Split(Path.DirectorySeparatorChar).Last(), substring using GetParent()?

Reply:
If we do a substring @derrickstolee had the suggestion of 
directoryPath.TrimEnd(Path.DirectorySeparatorChar).Substring(directoryPath.LastIndexOf(separator) + 1)

Shame there isn't a builtin function to do this but I'm not really bothered with which permuation we go with.

Reply:
@derrickstolee I think I can use that to simplify